Sources/Brave/Frontend/Browser/BrowserViewController.swift

```diff
@@ -3344,3 +3383,70 @@ extension BrowserViewController {
     self.present(host, animated: true)
   }
 }
+
+// Certificate info
+extension BrowserViewController {
+  
+  func displayPageCertificateInfo() {
+    guard let webView = tabManager.selectedTab?.webView else {
+      Logger.module.error("Invalid WebView")
+      return
+    }
+    
+    let getServerTrustForErrorPage = { () -> SecTrust? in
+      do {
+        if let url = webView.url {
+          return try ErrorPageHelper.serverTrust(from: url)
+        }
+      } catch {
+        Logger.module.error("\(error.localizedDescription)")
+      }
+      
+      return nil
+    }
+    
+    guard let trust = webView.serverTrust ?? getServerTrustForErrorPage() else {
+      return
+    }
+    
+    let host = webView.url?.host
+    
+    Task.detached {
+      let serverCertificates: [SecCertificate] = SecTrustCopyCertificateChain(trust) as? [SecCertificate] ?? []
+      
+      // TODO: Instead of showing only the first cert in the chain,
+      // have a UI that allows users to select any certificate in the chain (similar to Desktop browsers)
+      if let serverCertificate = serverCertificates.first,
```

Comment:
what does `first` certificate mean? if it's the leaf cert, that's fine. if it's any kind of cert higher up in the chain, i don't think we should show it at all since it would just confuse people. 

for instance on brave.com, the leaf cert says `Common Name: Brave.com`, the intermediate cert is for Amazon RSA 2048 M01, and the root cert is possibly something else.

Reply:
This is just moving the code that was already set up for showing the certificate viewer, so maybe @Brandon-T can clarify, but I assume that its the first in the chain and that results in the leaf always. Brandon has some further changes to actually view _all_ the certs in the chain but that was separate work

Reply:
Yes it is the leaf certificate. My PR has some changes that let the user see ALL certificates (leaf, intermediates, root) and they can select which one they want to see (though it shows leaf by default).

But the very first certificate is the leaf, and the root is the last (Apple always returns them in the reverse order).

Reply:
sgtm thanks!